flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1073,57 +1073,73 @@ class JobManager(
       // execute the recovery/writing the jobGraph into the SubmittedJobGraphStore asynchronously
       // because it is a blocking operation
       future {
-        try {
-          if (isRecovery) {
-            executionGraph.restoreLatestCheckpointedState()
-          }
-          else {
-            val snapshotSettings = jobGraph.getSnapshotSettings
-            if (snapshotSettings != null) {
-              val savepointPath = snapshotSettings.getSavepointPath()
+        val restoreStateSuccess =
+          try {
+            if (isRecovery) {
+              executionGraph.restoreLatestCheckpointedState()
```

Comment:
I'm not so sure about that, to be honest. What if the `restoreLatestCheckpointedState` fails because of some HDFS/ZooKeeper problems. Then you would like to try restarting the job, wouldn't you? The client should then be notified once all restarting attempts have been exhausted.